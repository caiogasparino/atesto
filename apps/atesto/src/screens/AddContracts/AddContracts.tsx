// import { useEffect } from 'react';
import { Field, Formik } from 'formik';
import * as Yup from 'yup';

import { Button, CircularProgress } from '@mui/material';

import Header from '../../components/Header/Header';
import { usePostContract } from '../../hooks/usePostContract';

import { StyledForm } from './styles';

interface FormValues {
  fornecedorId: number;
  nrContrato: string;
  pregao: string;
  valorContrato: number;
  gestorContratoId: number;
  fiscalContratoId: number;
  itens: {
    descItem: string;
    quantidade: number;
    valorUnitario: number;
    valorTotal: number;
    lote: string;
  }[];
}

const initialValues: FormValues = {
  fornecedorId: 0,
  nrContrato: '',
  pregao: '',
  valorContrato: 0,
  gestorContratoId: 1,
  fiscalContratoId: 1,
  itens: [
    {
      descItem: '',
      quantidade: 0,
      valorUnitario: 0,
      valorTotal: 0,
      lote: '',
    },
  ],
};

const validationSchema = Yup.object({
  fornecedorId: Yup.number().required(
    'O campo "ID do fornecedor" é obrigatório'
  ),
  nrContrato: Yup.string().required(
    'O campo "Número do contrato" é obrigatório'
  ),
  pregao: Yup.string().required('O campo "Pregão" é obrigatório'),
  valorContrato: Yup.number().required(
    'O campo "Valor do contrato" é obrigatório'
  ),
  gestorContratoId: Yup.number().required(
    'O campo "ID do gestor do contrato" é obrigatório'
  ),
  fiscalContratoId: Yup.number().required(
    'O campo "ID do fiscal do contrato" é obrigatório'
  ),
  itens: Yup.array().of(
    Yup.object().shape({
      descItem: Yup.string().required(
        'O campo "Descrição do item" é obrigatório'
      ),
      quantidade: Yup.number().required('O campo "Quantidade" é obrigatório'),
      valorUnitario: Yup.number().required(
        'O campo "Valor unitário" é obrigatório'
      ),
      valorTotal: Yup.number().required('O campo "Valor total" é obrigatório'),
      lote: Yup.string().required('O campo "Lote" é obrigatório'),
    })
  ),
});

const supplier = [
  {
    id: 1,
    cnpj: '81243735000148',
    razaoSocial: 'Teste de Update dddddddwwwwWWWWWWWWWddddd',
    telefone: '4133337886',
  },
  {
    id: 8,
    cnpj: '33050071000158',
    razaoSocial: 'Razão Social - Insert via API',
    telefone: '4133333333',
  },
  {
    id: 16,
    cnpj: '33050196000188',
    razaoSocial: 'Razão Social - Insert via API',
    telefone: '4133333333',
  },
  {
    id: 21,
    cnpj: '06981180000116',
    razaoSocial: 'Teste de Update dddddddddddd',
    telefone: '4133337886',
  },
  {
    id: 28,
    cnpj: '29978814000187',
    razaoSocial: 'teste 04/02/2023',
    telefone: '4144444444',
  },
  {
    id: 29,
    cnpj: '58119199000151',
    razaoSocial: 'string',
    telefone: '4133333333',
  },
  {
    id: 31,
    cnpj: '20258278000170',
    razaoSocial: 'string',
    telefone: '4133333333',
  },
  {
    id: 32,
    cnpj: '44380196000162',
    razaoSocial: 'string',
    telefone: '0099882233',
  },
  {
    id: 34,
    cnpj: '00000000000000',
    razaoSocial: 'Posi',
    telefone: '4199999999',
  },
];

export const AddContracts = () => {
  const { loading, postContract } = usePostContract();

  const handleSubmit = (values: FormValues) => {
    const data = postContract(values);
    return data;
  };
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={handleSubmit}
    >
      {(formik) => (
        <div>
          <Header title="ADICIONAR CONTRATOS" />
          <StyledForm>
            <form onSubmit={formik.handleSubmit}>
              <div>
                <label htmlFor="fornecedorId">ID do fornecedor</label>
                <select id="status" {...formik.getFieldProps('fornecedorId')}>
                  {supplier.map((itens, index) => (
                    <option value={index}>{itens.razaoSocial}</option>
                  ))}
                </select>
                {formik.touched.fornecedorId && formik.errors.fornecedorId && (
                  <div>{formik.errors.fornecedorId}</div>
                )}
              </div>

              <div>
                <label htmlFor="nrContrato">Número do contrato</label>
                <Field type="text" id="nrContrato" name="nrContrato" />
                {formik.touched.nrContrato && formik.errors.nrContrato && (
                  <div>{formik.errors.nrContrato}</div>
                )}
              </div>

              <div>
                <label htmlFor="pregao">Pregão</label>
                <Field type="text" id="pregao" name="pregao" />
                {formik.touched.pregao && formik.errors.pregao && (
                  <div>{formik.errors.pregao}</div>
                )}
              </div>
              <div>
                <label htmlFor="valorContrato">Valor do contrato</label>
                <Field type="number" id="valorContrato" name="valorContrato" />
                {formik.touched.valorContrato &&
                  formik.errors.valorContrato && (
                    <div>{formik.errors.valorContrato}</div>
                  )}
              </div>

              <div>
                <label htmlFor="gestorContratoId">
                  ID do gestor do contrato
                </label>
                <Field
                  type="number"
                  id="gestorContratoId"
                  name="gestorContratoId"
                />
                {formik.touched.gestorContratoId &&
                  formik.errors.gestorContratoId && (
                    <div>{formik.errors.gestorContratoId}</div>
                  )}
              </div>

              <div>
                <label htmlFor="fiscalContratoId">
                  ID do fiscal do contrato
                </label>
                <Field
                  type="number"
                  id="fiscalContratoId"
                  name="fiscalContratoId"
                />
                {formik.touched.fiscalContratoId &&
                  formik.errors.fiscalContratoId && (
                    <div>{formik.errors.fiscalContratoId}</div>
                  )}
              </div>

              <div>
                <label htmlFor="itens[0].descItem">Descrição do item</label>
                <Field
                  type="text"
                  id="itens[0].descItem"
                  name="itens[0].descItem"
                />
              </div>

              <div>
                <label htmlFor="itens[0].quantidade">Quantidade</label>
                <Field
                  type="number"
                  id="itens[0].quantidade"
                  name="itens[0].quantidade"
                />
              </div>

              <div>
                <label htmlFor="itens[0].valorUnitario">Valor unitário</label>
                <Field
                  type="number"
                  id="itens[0].valorUnitario"
                  name="itens[0].valorUnitario"
                />
              </div>

              <div>
                <label htmlFor="itens[0].valorTotal">Valor total</label>
                <Field
                  type="number"
                  id="itens[0].valorTotal"
                  name="itens[0].valorTotal"
                />
              </div>

              <div>
                <label htmlFor="itens[0].lote">Lote</label>
                <Field type="text" id="itens[0].lote" name="itens[0].lote" />
              </div>
            </form>
            <button type="submit" disabled={!loading}>
              {loading ? <CircularProgress /> : 'ENVIAR'}
            </button>
          </StyledForm>
        </div>
      )}
    </Formik>
  );
};
