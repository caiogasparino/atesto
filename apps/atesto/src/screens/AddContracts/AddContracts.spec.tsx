import { render } from '@testing-library/react';

import { AddContracts } from './AddContracts';

describe('AddContracts', () => {
  describe('AddContracts Snap', () => {
    it('renders correctly', () => {
      const { container } = render(<AddContracts />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
  it('should render successfully', () => {
    const { baseElement } = render(<AddContracts />);
    expect(baseElement).toBeTruthy();
  });
});
