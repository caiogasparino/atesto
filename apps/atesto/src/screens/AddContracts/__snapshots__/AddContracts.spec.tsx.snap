// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContracts AddContracts renders correctly 1`] = `
<div>
  <div
    class="Header__HeaderContainer-uzqoib-0 kaGncl"
  >
    <div
      class="Header__ButtonIconLeft-uzqoib-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          width="80"
        >
          icon_home.svg
        </svg>
        <h1
          class="Header__Text-uzqoib-5 dgxrWf"
        >
          INÍCIO
        </h1>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="Header__HeaderTitle-uzqoib-2"
    >
      <h1
        class="Header__Title-uzqoib-4 cZotme"
      >
        ADICIONAR CONTRATOS
      </h1>
    </div>
    <div
      class="Header__ButtonIconRight-uzqoib-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          width="80"
        >
          icon_arrow_circle_left.svg
        </svg>
        <h1
          class="Header__Text-uzqoib-5 dgxrWf"
        >
          VOLTAR
        </h1>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <form
    action="#"
    class="styles__StyledForm-nufrg-0 dHvgSY"
  >
    <div>
      <label
        for="numNFV"
      >
        Número NFV
      </label>
      <input
        id="numNFV"
        name="numNFV"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="serie"
      >
        Série
      </label>
      <input
        id="serie"
        name="serie"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="codigo"
      >
        Código
      </label>
      <input
        id="codigo"
        name="codigo"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="dataEmissao"
      >
        Data de Emissão
      </label>
      <input
        id="dataEmissao"
        lang="pr-Br"
        name="dataEmissao"
        type="datetime-local"
        value="Thu Apr 13 2023 19:42:32 GMT-0300 (Brasilia Standard Time)"
      />
    </div>
    <div>
      <label
        for="valorNota"
      >
        Valor da Nota
      </label>
      <input
        id="valorNota"
        name="valorNota"
        type="number"
        value="0"
      />
    </div>
    <div>
      <label
        for="status"
      >
        Status
      </label>
      <select
        id="status"
        name="status"
      >
        <option
          value="0"
        >
          Pendente
        </option>
        <option
          value="1"
        >
          Pago
        </option>
        <option
          value="2"
        >
          Cancelado
        </option>
      </select>
    </div>
    <div>
      <label
        for="contratoId"
      >
        ID do Contrato
      </label>
      <input
        id="contratoId"
        name="contratoId"
        type="number"
        value="0"
      />
    </div>
    <div>
      <label
        for="fornecedorId"
      >
        ID do Fornecedor
      </label>
      <input
        id="fornecedorId"
        name="fornecedorId"
        type="number"
        value="0"
      />
    </div>
    <div>
      <label
        for="workplaceId"
      >
        ID do Local de Trabalho
      </label>
      <input
        id="workplaceId"
        name="workplaceId"
        type="number"
        value="0"
      />
    </div>
    <button
      type="submit"
    >
      Enviar
    </button>
  </form>
</div>
`;
